politeiawww/cmd/pictl/cmdcommenttimestamps.go

```diff
@@ -34,23 +34,80 @@ func (c *cmdCommentTimestamps) Execute(args []string) error {
 		return err
 	}
 
-	// Get timestamps
-	t := cmv1.Timestamps{
-		Token:      c.Args.Token,
-		CommentIDs: c.Args.CommentIDs,
+	// If given comment IDs is empty fetch all IDs and request timestamps
+	// page by page.
+	commentIDs := c.Args.CommentIDs
+	if len(commentIDs) == 0 {
+		// Fetch all comments
+		cm := cmv1.Comments{
+			Token: c.Args.Token,
+		}
+		cmr, err := pc.Comments(cm)
+		if err != nil {
+			return err
+		}
+
+		// Collect comment IDs
+		for _, c := range cmr.Comments {
+			commentIDs = append(commentIDs, c.CommentID)
+		}
 	}
-	tr, err := pc.CommentTimestamps(t)
-	if err != nil {
-		return err
+
+	// If the proposal has no comments yet, nothing to do.
+	if len(commentIDs) == 0 {
+		printf("Proposal has no comments yet \n")
+		return nil
 	}
 
-	// Verify timestamps
-	notTimestamped, err := pclient.CommentTimestampsVerify(*tr)
+	// Get timestamps page size
+	pr, err := pc.CommentPolicy()
 	if err != nil {
 		return err
 	}
+	pageSize := pr.TimestampsPageSize
+
+	// Timestamps route is paginated, request timestamps page by page.
+	var (
+		pageStartIdx int
+		fetched      int
+	)
+	for pageStartIdx < len(commentIDs) {
+		pageEndIdx := pageStartIdx + int(pageSize)
+		if pageEndIdx > len(commentIDs) {
+			// We've reached the end of the slice
+			pageEndIdx = len(commentIDs)
+		}
+
+		// pageStartIdx is included. pageEndIdx is excluded.
+		page := commentIDs[pageStartIdx:pageEndIdx]
+
+		// Get timestamps
+		t := cmv1.Timestamps{
+			Token:      c.Args.Token,
+			CommentIDs: page,
+		}
+		tr, err := pc.CommentTimestamps(t)
+		if err != nil {
+			return err
+		}
+		fetched = fetched + len(page)
+
+		// Verify timestamps
+		notTimestamped, err := pclient.CommentTimestampsVerify(*tr)
+		if err != nil {
+			return err
+		}
+		if len(notTimestamped) > 0 {
+			printf("Not timestamped yet: %v\n", notTimestamped)
+		}
+
+		printf("Fetched timestampes of %v/%v comments \n", fetched,
```

Comment:
currently we don't print the returned results, I guess the reason is that the returned struct is can have several timestamps for each comment:

```
type CommentTimestamp struct {
	Adds []Timestamp `json:"adds"`
	Del  *Timestamp  `json:"del,omitempty"`
}
```

Maybe we could print the `Del` timestamp is exists and if not just print the last timestamp in the `Adds` 
slice, what do you think @lukebp ?

Reply:
You can print then by using the --json or --verbose flag, but we don't print them by default because it's a lot of data and not really helpful to print it all to stdout by default.  I'd prefer to print some kind of summary statistics about the timestamps by default rather than the timestamps themselves.

Reply:
updated:

```
```
$ pictl commenttimestamps 4178f2f 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 --timer
Total number of comments: 100, fetched: 100, timestamped: 100, not timestamped: 0
```
```